From 86e10d423662c9c5a208afcc81b8e12f17828a4d Mon Sep 17 00:00:00 2001
From: Hidehiko Abe <hidehiko@chromium.org>
Date: Tue, 4 Sep 2018 20:06:42 +0900
Subject: [PATCH] Manually cherry-pick SplitStringUsingSubstr.

https://codereview.chromium.org/2356253006 changed the signature.
To uprev libchrome step-by-step, this CL manually cherry-pick
SplitStringUsingSubstr from the CL.

BUG=b:37434548
TEST=Trybot.

Change-Id: I4cdc475e072b14e456d5eecf73f325b5baddb9a7
---
 base/strings/string_split.cc | 18 ++++++++++++++++++
 base/strings/string_split.h  | 17 +++++++++++++----
 2 files changed, 31 insertions(+), 4 deletions(-)

diff --git a/base/strings/string_split.cc b/base/strings/string_split.cc
index 6c949b9..64d9c58 100644
--- a/base/strings/string_split.cc
+++ b/base/strings/string_split.cc
@@ -227,6 +227,24 @@ bool SplitStringIntoKeyValuePairs(StringPiece input,
   return success;
 }
 
+std::vector<string16> SplitStringUsingSubstr(StringPiece16 input,
+                                             StringPiece16 delimiter,
+                                             WhitespaceHandling whitespace,
+                                             SplitResult result_type) {
+  std::vector<string16> result;
+  SplitStringUsingSubstrT(input, delimiter, whitespace, result_type, &result);
+  return result;
+}
+
+std::vector<std::string> SplitStringUsingSubstr(StringPiece input,
+                                                StringPiece delimiter,
+                                                WhitespaceHandling whitespace,
+                                                SplitResult result_type) {
+  std::vector<std::string> result;
+  SplitStringUsingSubstrT(input, delimiter, whitespace, result_type, &result);
+  return result;
+}
+
 void SplitStringUsingSubstr(StringPiece16 input,
                             StringPiece16 delimiter,
                             std::vector<string16>* result) {
diff --git a/base/strings/string_split.h b/base/strings/string_split.h
index ec9f246..971df29 100644
--- a/base/strings/string_split.h
+++ b/base/strings/string_split.h
@@ -90,10 +90,19 @@ BASE_EXPORT bool SplitStringIntoKeyValuePairs(StringPiece input,
 
 // Similar to SplitString, but use a substring delimiter instead of a list of
 // characters that are all possible delimiters.
-//
-// TODO(brettw) this should probably be changed and expanded to provide a
-// mirror of the SplitString[Piece] API above, just with the different
-// delimiter handling.
+BASE_EXPORT std::vector<string16> SplitStringUsingSubstr(
+    StringPiece16 input,
+    StringPiece16 delimiter,
+    WhitespaceHandling whitespace,
+    SplitResult result_type);
+BASE_EXPORT std::vector<std::string> SplitStringUsingSubstr(
+    StringPiece input,
+    StringPiece delimiter,
+    WhitespaceHandling whitespace,
+    SplitResult result_type);
+
+// These are kept for backword compatibility.
+// TODO(hidehiko): Remove these on uprev to r456626.
 BASE_EXPORT void SplitStringUsingSubstr(StringPiece16 input,
                                         StringPiece16 delimiter,
                                         std::vector<string16>* result);
-- 
2.19.0.rc1.350.ge57e33dbd1-goog

